docs/src/content/en/guides/build-your-ui/ai-sdk-ui.mdx

```diff
@@ -450,6 +450,98 @@ for await (const part of uiMessageStream) {
 }
 ```
 
+## Using Mastra Features with AI SDK
+
+The `withMastra()` function wraps an AI SDK model with Mastra features, allowing you to use [processors](/docs/v1/agents/processors) and [memory](/docs/v1/memory/memory-processors) with AI SDK's `generateText`, `streamText`, `generateObject`, and `streamObject` functions.
```

Comment:
An explanation of why you might wanna do this (instead of using Mastra directly) would be cool